File: technical_skills_java/src/main/java/com/natiqhaciyef/technical_skills_java/leetcode/interview/MergeSortedArrays.java
package com.natiqhaciyef.technical_skills_java.leetcode.interview;

import java.util.ArrayList;

public class MergeSortedArrays {
    public static void merge(int[] nums1, int m, int[] nums2, int n) {
        for (int i = 0; i < nums2.length; i++) {
            if (nums2[i] != 0) {
                for (int j = 0; j < nums1.length; j++) {
                    if (nums1[j] == 0) {
                        nums1[j] = nums2[i];
                        break;
                    }
                }
            }
        }

        // insertion sort

        System.out.print("[");
        for (int e = 0; e < nums1.length; e++) {
            if (e < nums1.length - 1)
                System.out.print(nums1[e] + ",");
            else
                System.out.print(nums1[e]);
        }

        System.out.print("]");
    }

    public static void main(String[] args) {
        int[] arr1 = {-1, 0, 0, 3, 3, 3, 0, 0, 0};
        int[] arr2 = {1, 2, 2};
        merge(arr1, 6, arr2, 3);
    }
}

File: technical_skills_kotlin/src/main/java/com/natiqhaciyef/technical_skills_kotlin/leetcode/interview/Merge Sorted Arrays.kt
package com.natiqhaciyef.technical_skills_kotlin.leetcode.interview


fun merge(nums1: IntArray, m: Int, nums2: IntArray, n: Int) {
    for (element in nums2) {
        if (element > 0) {
            val index = nums1.indexOf(0)
            nums1[index] = element
        }
    }

    // insertion sort
    for (k in 1 until nums1.size) {
        val key = nums1[k]
        var j = k - 1

        while (j >= 0 && nums1[j] > key) {
            nums1[j + 1] = nums1[j] //(j+1 <= k)
            j-- // j - * loop-count
        }

        nums1[j + 1] = key
    }

    print("[")
    nums1.forEachIndexed { index, i ->
        if (nums1.lastIndex != index)
            print("${nums1[index]}, ")
        else
            print("${nums1[index]}")
    }

    print("]")
}

fun main() {
    val arr1 = intArrayOf(0, 17, 6, 2, 0, 5, 0, 0, 4, 6, 3, 0, 0, 0)
    val arr2 = intArrayOf(12, 2, 5, 8, 4, 3, 8, 0)
    merge(arr1, arr1.size - 6, arr2, arr2.size - 1)
}